File: src/main/java/me/zeroeightsix/kami/module/modules/sdashb/combat/AntiDeathScreen.java
package me.zeroeightsix.kami.module.modules.sdashb.combat;

import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.setting.Setting;
import me.zeroeightsix.kami.setting.Settings;
import net.minecraft.client.gui.GuiGameOver;

@Module.Info(name = "AntiDeathScreen", description = "Fixs random death screen glitches", category = Module.Category.COMBAT)
public class AntiDeathScreen extends Module {

    //private Setting<Boolean> respawn = this.register(Settings.b("Respawn", true));

    @Override
    public void onUpdate()
    {
        if (!this.isEnabled()) return;
        if (mc.player.getHealth() > 0 && mc.currentScreen instanceof GuiGameOver)
        {
            mc.player.respawnPlayer();
            mc.displayGuiScreen(null);
        }
    }

}

File: src/main/kotlin/com/lambda/client/module/modules/combat/AntiDeathScreen.kt
package com.lambda.client.module.modules.combat

import com.lambda.client.event.events.GuiEvent
import com.lambda.client.module.Category
import com.lambda.client.module.Module
import com.lambda.client.util.threads.safeListener
import net.minecraft.client.gui.GuiGameOver

object AntiDeathScreen : Module(
    name = "AntiDeathScreen",
    description = "Fixes random death screen glitches",
    category = Category.COMBAT
) {
    init {
        safeListener<GuiEvent.Displayed> {
            if (it.screen !is GuiGameOver) return@safeListener
            if (player.health > 0) {
                player.respawnPlayer()
                mc.displayGuiScreen(null)
            }
        }
    }
}